opensearch/src/main/java/org/opensearch/sql/opensearch/client/OpenSearchNodeClient.java

```diff
@@ -50,6 +53,28 @@ public OpenSearchNodeClient(NodeClient client) {
     this.resolver = new IndexNameExpressionResolver(client.threadPool().getThreadContext());
   }
 
+  @Override
+  public boolean exists(String indexName) {
+    try {
+      IndicesExistsResponse checkExistResponse = client.admin().indices()
+          .exists(new IndicesExistsRequest(indexName)).actionGet();
```

Comment:
Question, i'm seeing other plugins using [routing table](https://javadoc.io/static/org.elasticsearch/elasticsearch/7.10.2/org/elasticsearch/cluster/routing/RoutingTable.html#hasIndex(java.lang.String)) to get index, for example [alerting](https://github.com/opensearch-project/alerting/blob/5074f7b38dfb3039a499c6c8c46094e513cccace//alerting/src/main/kotlin/org/opensearch/alerting/alerts/AlertIndices.kt#L257). Would that be faster or any other difference?

Reply:
> Question, i'm seeing other plugins using [routing table](https://javadoc.io/static/org.elasticsearch/elasticsearch/7.10.2/org/elasticsearch/cluster/routing/RoutingTable.html#hasIndex(java.lang.String)) to get index, for example [alerting](https://github.com/opensearch-project/alerting/blob/5074f7b38dfb3039a499c6c8c46094e513cccace//alerting/src/main/kotlin/org/opensearch/alerting/alerts/AlertIndices.kt#L257). Would that be faster or any other difference?

I guess they're trading off consistency for performance by checking state in `clusterState` object. I can see some `exists` API call in alerting code with `local=true`. There was similar code in SQL reading index mapping in cluster state to avoid blocking I/O. I will double check as I keep working on this feature. Thanks for pointing out this!